src/Rector/v12/v0/typo3/AbstractMessageGetSeverityRector.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v12\v0\typo3;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/12.0/Breaking-97787-AbstractMessageGetSeverityReturnsContextualFeedbackSeverity.html
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\AbstractMessageGetSeverityRector\AbstractMessageGetSeverityRectorTest
+ */
+final class AbstractMessageGetSeverityRector extends AbstractRector
+{
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, PropertyFetch::class];
+    }
+
+    /**
+     * @param MethodCall|PropertyFetch $node
+     */
+    public function refactor(Node $node)
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        if (! $this->isNames($node->name, ['getSeverity'])) {
+            return null;
+        }
+
+        return null;
```

Comment:
this last segment should then actually apply the change
i would expect something like
`return $this->nodeFactory->createPropertyFetch($node, 'value')`
or maybe `$node->var`, not sure